core-primitives/enclave-api/src/remote_attestation.rs

```diff
@@ -131,306 +118,343 @@ pub trait TlsRemoteAttestation {
 	) -> EnclaveResult<()>;
 }
 
-impl RemoteAttestation for Enclave {
-	fn generate_ias_ra_extrinsic(&self, w_url: &str, skip_ra: bool) -> EnclaveResult<Vec<u8>> {
-		let mut retval = sgx_status_t::SGX_SUCCESS;
-
-		let mut unchecked_extrinsic: Vec<u8> = vec![0u8; EXTRINSIC_MAX_SIZE];
-
-		trace!("Generating dcap_ra_extrinsic with URL: {}", w_url);
-
-		let url = w_url.encode();
-
-		let result = unsafe {
-			ffi::generate_ias_ra_extrinsic(
-				self.eid,
-				&mut retval,
-				url.as_ptr(),
-				url.len() as u32,
-				unchecked_extrinsic.as_mut_ptr(),
-				unchecked_extrinsic.len() as u32,
-				skip_ra.into(),
-			)
-		};
-
-		ensure!(result == sgx_status_t::SGX_SUCCESS, Error::Sgx(result));
-		ensure!(retval == sgx_status_t::SGX_SUCCESS, Error::Sgx(retval));
+#[cfg(feature = "implement-ffi")]
+mod impl_ffi {
```

Comment:
Huge diff, but I simply nested the implementation into a separate module, because of all the imports, which I didn't want to feature gate individually. Same is true for the other files.